src/clustering/administration/servers/auto_reconnect.cc

```diff
@@ -87,7 +87,18 @@ void auto_reconnector_t::try_reconnect(const server_id_t &server,
     exponential_backoff_t backoff(50, 15 * 1000);
     try {
         while (!interruptor.is_pulsed()) {
-            connectivity_cluster_run->join(last_known_address, join_delay_secs);
+            join_result_t result =
+                connectivity_cluster_run->join_blocking(
+                    last_known_address, boost::none, server,
+                    join_delay_secs,
+                    auto_drainer_t::lock_t(&drainer));
```

Comment:
simply removing the guarantee leads to other crazier errors, so I think we have some logical errors in here as well. I'll wait to hear back from you.